src/Driver.cpp

```diff
@@ -1072,11 +1067,12 @@ bool Driver::writePacket(uint8_t const* buffer, int buffer_size, int timeout)
             return true;
         }
 
-        if (time_out.elapsed())
-            throw TimeoutError(TimeoutError::PACKET, "writePacket(): timeout");
-
         int remaining_timeout = time_out.timeLeft();
-        m_stream->waitWrite(Time::fromMicroseconds(remaining_timeout * 1000));
+
+        if (time_out.elapsed() || !m_stream->waitWrite(Time::fromMicroseconds(remaining_timeout * 1000))) 
+        {
+            throw TimeoutError(TimeoutError::PACKET, "writePacket(): timeout");            
```

Comment:
waitWrite originally threw TimeoutError::NONE ... I would rather keep the behavior identical within this pull request.

Reply:
It was a bit inconsistent: writepacket threw TimeoutError::PACKET if the timeout has elapsed before calling waitwrite, but otherwise forwarded the TimeoutError::NONE from waitwrite. So do you want to discriminate the two cases again?
What is the supposed meaning of TimeoutError::NONE, btw?

Reply:
In principle, I would welcome a cleaning up of this. However, I'd prefer we do it step by step.

- `NONE` should be `UNKNOWN` IMO. To be honest, it looks like it was created for the purpose of waitRead/waitWrite because these two methods could not know what to throw - this change will improve that logic as well (which is good !)
- in principle, if we were to follow the nomenclature of the read side, we should probably use FIRST_BYTE if no bytes were written and PACKET otherwise (i.e. get rid of the NONE).

But, as I said, I'd prefer we keep the two things separate. iodrivers_base is probably one of the most widely used Rock library ... I'd rather change things a step at a time.